test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -83,95 +90,163 @@ var _ = AfterSuite(func() {
+				curlCMD := fmt.Sprintf("curl --max-time 60 %s", dashboardURL)
+				output, err := exec.Command("ssh", "-i", sshKeyPath, "-o", "ConnectTimeout=10", "-o", "StrictHostKeyChecking=no", "-o", "UserKnownHostsFile=/dev/null", master, curlCMD).CombinedOutput()
+				if err != nil {
+					log.Printf("\n\nOutput:%s\n\n", string(output))
+				}
+			}
+		})
+	})
+
+	Context("with a linux agent pool", func() {
+		It("should be able to deploy an nginx service", func() {
+			if eng.HasLinuxAgents() {
+				r := rand.New(rand.NewSource(time.Now().UnixNano()))
+				deploymentName := fmt.Sprintf("%s-%v", cfg.Name, r.Intn(99999))
+				d, err := deployment.CreateLinuxDeploy("library/nginx:latest", deploymentName, "default")
+				Expect(err).NotTo(HaveOccurred())
+				err = d.Expose(80)
+				Expect(err).NotTo(HaveOccurred())
+
+				s, err := service.Get(deploymentName, "default")
+				Expect(err).NotTo(HaveOccurred())
+				s, err = s.WaitForExternalIP(360, 5)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(s.Status.LoadBalancer.Ingress).NotTo(BeEmpty())
+
+				url := fmt.Sprintf("http://%s", s.Status.LoadBalancer.Ingress[0]["ip"])
+				resp, err := http.Get(url)
+				Expect(err).NotTo(HaveOccurred())
+				defer resp.Body.Close()
+				body, err := ioutil.ReadAll(resp.Body)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(string(body)).To(MatchRegexp("(Welcome to nginx)"))
+			} else {
+				Skip("No linux agent was provisioned for this Cluster Definition")
+			}
+		})
 	})
 
+	Context("with a windows agent pool", func() {
+		It("should be able to deploy a powershell webserver", func() {
+			if eng.HasWindowsAgents() {
+				command := `powershell.exe -command "<#code used from https://gist.github.com/wagnerandrade/5424431#> ; $$listener = New-Object System.Net.HttpListener ; $$listener.Prefixes.Add('http://*:80/') ; $$listener.Start() ; $$callerCounts = @{} ; Write-Host('Listening at http://*:80/') ; while ($$listener.IsListening) { ;$$context = $$listener.GetContext() ;$$requestUrl = $$context.Request.Url ;$$clientIP = $$context.Request.RemoteEndPoint.Address ;$$response = $$context.Response ;Write-Host '' ;Write-Host('> {0}' -f $$requestUrl) ;  ;$$count = 1 ;$$k=$$callerCounts.Get_Item($$clientIP) ;if ($$k -ne $$null) { $$count += $$k } ;$$callerCounts.Set_Item($$clientIP, $$count) ;$$header='<html><body><H1>Windows Container Web Server</H1>' ;$$callerCountsString='' ;$$callerCounts.Keys | % { $$callerCountsString+='<p>IP {0} callerCount {1} ' -f $$_,$$callerCounts.Item($$_) } ;$$footer='</body></html>' ;$$content='{0}{1}{2}' -f $$header,$$callerCountsString,$$footer ;Write-Output $$content ;$$buffer = [System.Text.Encoding]::UTF8.GetBytes($$content) ;$$response.ContentLength64 = $$buffer.Length ;$$response.OutputStream.Write($$buffer, 0, $$buffer.Length) ;$$response.Close() ;$$responseStatus = $$response.StatusCode ;Write-Host('< {0}' -f $$responseStatus)  } ; "`
+				r := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
Its from the `simpleweb-windows` deployment manifest. I would rather not rely on the external file if I can help it so doing this seemed like a better compromise